internal/resources/metal/vlan/resource.go

```diff
@@ -81,96 +82,103 @@ func resourceMetalVlan() *schema.Resource {
+		Description: metalv1.PtrString(d.Get("description").(string)),
 	}
 	if metroOk {
-		createRequest.Metro = metroRaw.(string)
-		createRequest.VXLAN = vxlanRaw.(int)
+		createRequest.Metro = metalv1.PtrString(metroRaw.(string))
+		createRequest.Vxlan = metalv1.PtrInt32(int32(vxlanRaw.(int)))
 	}
 	if facOk {
 		createRequest.Facility = facRaw.(string)
 	}
-	vlan, _, err := client.ProjectVirtualNetworks.Create(createRequest)
+	vlan, _, err := client.VLANsApi.CreateVirtualNetwork(ctx, d.Get("project_id").(string)).VirtualNetworkCreateInput(createRequest).Execute()
 	if err != nil {
-		return equinix_errors.FriendlyError(err)
+		return diag.FromErr(equinix_errors.FriendlyError(err))
 	}
-	d.SetId(vlan.ID)
-	return resourceMetalVlanRead(d, meta)
+	d.SetId(vlan.GetId())
+	return resourceMetalVlanRead(ctx, d, meta)
 }
 
-func resourceMetalVlanRead(d *schema.ResourceData, meta interface{}) error {
-	meta.(*config.Config).AddModuleToMetalUserAgent(d)
-	client := meta.(*config.Config).Metal
+func resourceMetalVlanRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	meta.(*config.Config).AddModuleToMetalGoUserAgent(d)
+	client := meta.(*config.Config).Metalgo
 
-	vlan, _, err := client.ProjectVirtualNetworks.Get(d.Id(),
-		&packngo.GetOptions{Includes: []string{"assigned_to"}})
+	vlan, _, err := client.VLANsApi.GetVirtualNetwork(ctx, d.Id()).
+		Include([]string{"assigned_to"}).Execute()
 	if err != nil {
 		err = equinix_errors.FriendlyError(err)
 		if equinix_errors.IsNotFound(err) {
 			d.SetId("")
 			return nil
 		}
-		return err
+		return diag.FromErr(err)
 
 	}
 	d.Set("description", vlan.Description)
-	d.Set("project_id", vlan.Project.ID)
-	d.Set("vxlan", vlan.VXLAN)
-	d.Set("facility", vlan.FacilityCode)
+	d.Set("project_id", vlan.AssignedTo.GetId()) // assigned_to is a project but specced as an href
+	d.Set("vxlan", vlan.GetVxlan())
+	d.Set("facility", vlan.FacilityCode) // vlan spec does not include facility_code; should we remove it?
 	d.Set("metro", vlan.MetroCode)
 	return nil
 }
 
-func resourceMetalVlanDelete(d *schema.ResourceData, meta interface{}) error {
-	meta.(*config.Config).AddModuleToMetalUserAgent(d)
-	client := meta.(*config.Config).Metal
+func resourceMetalVlanDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	meta.(*config.Config).AddModuleToMetalGoUserAgent(d)
+	client := meta.(*config.Config).Metalgo
 
 	id := d.Id()
-	vlan, resp, err := client.ProjectVirtualNetworks.Get(id, &packngo.GetOptions{Includes: []string{"instances", "instances.network_ports.virtual_networks", "internet_gateway"}})
-	if equinix_errors.IgnoreResponseErrors(equinix_errors.HttpForbidden, equinix_errors.HttpNotFound)(resp, err) != nil {
-		return equinix_errors.FriendlyError(err)
+	vlan, resp, err := client.VLANsApi.GetVirtualNetwork(ctx, id).Include([]string{"instances", "instances.network_ports.virtual_networks", "internet_gateway"}).Execute()
```

Comment:
It's possible a different include would fix the issue here (at least partially; may need spec changes so the SDK can correctly identify types)
```suggestion
	vlan, resp, err := client.VLANsApi.GetVirtualNetwork(ctx, id).Include([]string{"instances", "virtual_networks", "internet_gateway"}).Execute()
```